frontend/packages/ux-editor/src/components/config/editModal/EditOptions/OptionTabs/EditTab/OptionListEditor/OptionListEditor.tsx

```diff
@@ -1,39 +1,42 @@
-import React, { createRef, useRef, useState } from 'react';
-import type { Option } from 'app-shared/types/Option';
+import React from 'react';
 import { useTranslation } from 'react-i18next';
-import {
-  StudioCodeListEditor,
-  StudioModal,
-  StudioSpinner,
-  StudioErrorMessage,
-  StudioAlert,
-  StudioProperty,
-} from '@studio/components';
-import type { CodeListEditorTexts } from '@studio/components';
+import { StudioSpinner, StudioErrorMessage, StudioAlert } from '@studio/components';
 import { useStudioEnvironmentParams } from 'app-shared/hooks/useStudioEnvironmentParams';
-import { useUpdateOptionListMutation } from 'app-shared/hooks/mutations/useUpdateOptionListMutation';
 import { useOptionListsQuery } from 'app-shared/hooks/queries/useOptionListsQuery';
-import { useOptionListEditorTexts } from '../../hooks/useOptionListEditorTexts';
-import { usePreviewContext } from 'app-development/contexts/PreviewContext';
 import type { IGenericEditComponent } from '../../../../../componentConfig';
 import type { SelectionComponentType } from '../../../../../../../types/FormComponent';
-import classes from './OptionListEditor.module.css';
+import { LibraryOptionsEditor } from './LibraryOptionsEditor';
+import { ManualOptionsEditor } from './ManualOptionsEditor';
 
-type OptionListEditorProps = {
+export type OptionListEditorProps = {
   optionsId: string;
-  label: string;
+  setComponentHasOptionList: (value: boolean) => void;
 } & Pick<IGenericEditComponent<SelectionComponentType>, 'component' | 'handleComponentChange'>;
 
 export function OptionListEditor({
   optionsId,
   component,
```

Comment:
We dont need to pass the optionsId as a separate prop as it is deduced from the component

Reply:
Completely agree

Reply:
I thought i had removed it, it will likely also be fixed in the single-option-list-fetch PR.